packages/other/utils/src/computeDataAttributes.ts

```diff
@@ -12,11 +12,15 @@ const computeDataAttributes = (
     return {};
   }
 
-  return Object.fromEntries(
-    Object.entries(dataAttributes).map((entry) => {
+  return Object.entries(dataAttributes).reduce<DataAttributes>(
```

Comment:
@blurbyte `Object.entries` has no IE support at all — what do you think about implementing something that will work in IE for now? We had 64 IE11 users last week across all `app.datacamp.com` visits.

Reply:
I checked a few weeks ago and learner hub didn't load at all in IE11, and campus is also broken - I think we have "unofficially" dropped support: https://datacamp.slack.com/archives/C03RWC7HP/p1618386768101900

I guess the biggest likely impact here is on enterprise - do you reckon there is a big fallout if we don't fix this?

Having said that, it's no problem if you want to support it - just flagging it 😁

Reply:
Shouldn't be a lot of work, probably worth doing it for that 64 users :). I am wondering how serious we are about dropping IE11 support?

Reply:
> "How serious are we about dropping IE11 support?

I think that's a really good question. To my mind — and @jdmunro can chime in here — but we've sort of hedged our bets by saying "We don't officially support IE11 and we've registered with the upgrade program", but that doesn't mean that:

1. We won't get any IE11 visitors
2. Any current / future customers who use IE11 now will definitely upgrade from IE11
3. Individual teams may not be provided with a ARR reason to add IE11 support to some area of the platform

For now, let's disregard 2 and 3. Let's also disregard what Learner Hub does or doesn't do, as that's just a piece of Atlas, rather than the whole product offering.

Atlas currently will boot correctly for IE11 users, but the individual microfrontends might fail. I took that decision because there's an issue with Sentry right now where it's not possible to create full encapsulation of a microfrontend meaning some errors "leak" up a level to the composer. Not ideal. Sentry claim to be working to fix this in v7, but we'll see.

As we're living in a world where a) we will almost certainly get IE11 visitors b) Sentry will generate errors for these users, the options seem to be:

1. Teams need to manage all kinds of different errors coming in from Sentry from IE11 users OR put in blanket filters on Sentry projects to exclude issues from IE11 users.
2. Put a hard block on IE11 users: literally throw up a block + upgrade page using our own infra/code. No IE11 user ever touches the apps.

I'd probably be in favour of 2 as it would involve the least work and would mean that teams don't have to replicate the same configs across projects current and new. But it needs alignment. Perhaps it should be taken up by the Performance TF as transpiling does often come down to a bundle size issue.

---

The other reason I raised the question about `fromEntries` (no support on 3 browsers we get visitors from) is: how opinionated should Waffles be regarding transpilation / browser support?

Each app should be transpiling for the same set of allowed browsers, but the reality is that each app is likely transpiling slightly differently if it's using browserslist queries such as `not dead` or `last X versions`. Also, as experimental features get released in the last 2 versions of say Chrome, it's possible code like that could sneak into Waffles and negatively affect users of previous versions of Chrome.

### Should the individual apps be able to safely transpile anything that comes in from Waffles?

If the answer is yes, then I could set Atlas Header to transpile Account Dropdown to work in all browsers we want to target. And this would be updated each time the app builds and deploys ("rolling forward with progress").

If not, then we're reliant on the Design System team either writing super vanilla JS or transpiling all components for the right browser targets ahead of time. Plus they'd need to update releases to those components every time there's a significant shift in the browsers that would be targeted by our browserslist?

I'm not sure of the current approach of Waffles @blurbyte but I wonder if it's something we should decide and document/communicate?